libbeat/filter/filter.go

```diff
@@ -0,0 +1,167 @@
+package filter
+
+import (
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type FilterCondition struct {
+}
+
+type DropFieldsConfig struct {
+	Fields []string `yaml:"fields"`
+}
+
+type IncludeFieldsConfig struct {
+	Fields []string `yaml:"fields"`
+}
+
+type FilterConfig struct {
+	DropFields    *DropFieldsConfig    `yaml:"drop_fields"`
+	IncludeFields *IncludeFieldsConfig `yaml:"include_fields"`
+}
+
+type FilterRule interface {
+	Filter(event *common.MapStr) *common.MapStr
```

Comment:
but filter normally just returns another common.MapStr. Due to MapStr just being a typedef for map no pointer is required. We can even return a tuple `(newEvent common.MapStr, drop bool)` to make intention more clear.